components/script/dom/gpudevice.rs

```diff
@@ -463,4 +468,64 @@ impl GPUDeviceMethods for GPUDevice {
         let pipeline_layout = receiver.recv().unwrap();
         GPUPipelineLayout::new(&self.global(), bind_group_layouts, pipeline_layout, valid)
     }
+
+    /// https://gpuweb.github.io/gpuweb/#dom-gpudevice-createbindgroup
+    fn CreateBindGroup(&self, descriptor: &GPUBindGroupDescriptor) -> DomRoot<GPUBindGroup> {
+        let alignment: u64 = 256;
+        let mut valid = descriptor.layout.bindings().len() == descriptor.bindings.len();
+
+        valid &= descriptor.bindings.iter().all(|bind| {
+            let buffer_size = bind.resource.buffer.size();
+            let resource_size = bind.resource.size.unwrap_or(buffer_size);
+            let length = bind.resource.offset.checked_add(resource_size);
+
+            length.is_some() &&
+            buffer_size >= length.unwrap() && // check buffer OOB
+            bind.resource.offset % alignment == 0 && // check alignment
+            bind.resource.offset < buffer_size && // on Vulkan offset must be less than size of buffer
+            descriptor.layout.bindings().iter().any(|layout_bind| {
+                let usage = BufferUsage::from_bits(bind.resource.buffer.usage()).unwrap();
+                let ty = match layout_bind.type_ {
+                    GPUBindingType::Storage_buffer  => BufferUsage::STORAGE,
+                    // GPUBindingType::Readonly_storage_buffer  => BufferUsage::STORAGE_READ,
```

Comment:
Why is this commented out?

Reply:
The `BufferUsage::STORAGE_READ` is not in the Draft so neither in the WebIDL yet, but it is in wgpu-core. I just wanted it to be in place already, but I can remove it.